### ISSUE TITLE
tools/jpackage/windows/Win8282351Test.java in jdk_tools_0 fails on most machines, passing on test-azure-win2012r2-x64-3

### ISSUE THREAD

**Issue by smlambert** [adoptium/infrastructure#2885]
tools/jpackage/windows/Win8282351Test.java (in jdk_tools target), fails in error on build-alibaba-win2012r2-x64-2, test-ibmcloud-win2012r2-x64-1, but passes on test-azure-win2012r2-x64-3:

```
ERROR: Expected [0]. Actual [1]: Check command [C:\Users\Jenkins\workspace\Test_openjdk17_hs_extended.openjdk_x86-64_windows_testList_1\openjdkbinary\j2sdk-image\bin\jpackage.exe --dest .\test\output --name Win8282351Test$-$$-$$$ --type msi --app-image .\test\appimage\Win8282351Test --win-menu --win-shortcut --verbose](12) exited with 0 code
[10:20:12.777] [  FAILED  ] Win8282351Test.test; checks=4
```

- test suite/name (e.g, BUILD_LIST, TARGET, CUSTOM_TARGET)?
- a link into recent `Test_` job on https://ci.adoptopenjdk.net which showed the failure
- **Hyperlink** to re-run in Grinder: 
- Is there an existing issue elsewhere covering this?
- Which machine(s) does it work on? test-azure-win2012r2-x64-3 (see [Grinder 6319](https://ci.adoptopenjdk.net/job/Grinder/6319) for success run), not sure what version of WiX toolkit is present on this machine.
- Which machine(s) does it fail on? Most.  deep history shows failing on build-alibaba-win2012r2-x64-2, test-ibmcloud-win2012r2-x64-1, build-ibmcloud-win2012r2-x64-2, build-azure-win2012r2-x64-2 `WiX 3.14.0.3910 detected`
- mentioned in https://github.com/adoptium/aqa-tests/issues/4240#issuecomment-1382741799

---

**Test Info**
Test Name: jdk_tools_0
Test Duration: 17 min 38 sec
Machine: build-alibaba-win2012r2-x64-2
TRSS link for the test output: https://trss.adoptium.net/output/test?id=63c219e9ed0b30773ba354a9

**Build Info**
Build Name: Test_openjdk17_hs_extended.openjdk_x86-64_windows_testList_1
Jenkins Build start time: Jan 13 2023, 06:26 pm
Jenkins Build URL: https://ci.adoptopenjdk.net/job/Test_openjdk17_hs_extended.openjdk_x86-64_windows_testList_1/76/
TRSS link for the build: https://trss.adoptium.net/allTestsInfo?buildId=63c218eaed0b30773ba34d89

**Java Version**
openjdk version "17.0.6" 2023-01-17
OpenJDK Runtime Environment Temurin-17.0.6+8 (build 17.0.6+8)
OpenJDK 64-Bit Server VM Temurin-17.0.6+8 (build 17.0.6+8, mixed mode, sharing)

**This test has been failed 2 times since Jan 12 2023, 06:16 am**
**Java Version when the issue first seen**
openjdk version "17.0.6" 2023-01-17
OpenJDK Runtime Environment Temurin-17.0.6+8 (build 17.0.6+8)
OpenJDK 64-Bit Server VM Temurin-17.0.6+8 (build 17.0.6+8, mixed mode, sharing)
Jenkins Build URL: https://ci.adoptopenjdk.net/job/Test_openjdk17_hs_extended.openjdk_x86-64_windows_testList_1/75/

The test failed on machine build-alibaba-win2012r2-x64-2 1 times 
The test failed on machine test-ibmcloud-win2012r2-x64-1 1 times 

[Rerun in Grinder](https://ci.adoptopenjdk.net/job/Grinder/parambuild/?SDK_RESOURCE=upstream&amp;TARGET=jdk_tools_0&amp;TEST_FLAG=&amp;UPSTREAM_TEST_JOB_NAME=Test_openjdk17_hs_extended.openjdk_x86-64_windows&amp;DOCKER_REQUIRED=false&amp;ACTIVE_NODE_TIMEOUT=0&amp;VENDOR_TEST_DIRS=&amp;EXTRA_DOCKER_ARGS=&amp;TKG_OWNER_BRANCH=adoptium%3Amaster&amp;OPENJ9_SYSTEMTEST_OWNER_BRANCH=eclipse%3Amaster&amp;PLATFORM=x86-64_windows&amp;GENERATE_JOBS=false&amp;KEEP_REPORTDIR=true&amp;PERSONAL_BUILD=false&amp;ADOPTOPENJDK_REPO=https%3A%2F%2Fgithub.com%2Fadoptium%2Faqa-tests.git&amp;LABEL=&amp;EXTRA_OPTIONS=&amp;CUSTOMIZED_SDK_URL=&amp;BUILD_IDENTIFIER=&amp;ADOPTOPENJDK_BRANCH=v0.9.6-release&amp;LIGHT_WEIGHT_CHECKOUT=false&amp;USE_JRE=false&amp;ARTIFACTORY_SERVER=&amp;KEEP_WORKSPACE=false&amp;USER_CREDENTIALS_ID=&amp;JDK_VERSION=17&amp;ITERATIONS=1&amp;VENDOR_TEST_REPOS=&amp;JDK_REPO=https%3A%2F%2Fgithub.com%2Fadoptium%2Fjdk17u&amp;OPENJ9_BRANCH=master&amp;OPENJ9_SHA=&amp;JCK_GIT_REPO=&amp;VENDOR_TEST_BRANCHES=&amp;UPSTREAM_JOB_NAME=build-scripts%2Fjobs%2Frelease%2Fjobs%2Fjdk17u%2Fjdk17u-release-windows-x64-temurin&amp;OPENJ9_REPO=https%3A%2F%2Fgithub.com%2Feclipse-openj9%2Fopenj9.git&amp;CLOUD_PROVIDER=&amp;CUSTOM_TARGET=&amp;VENDOR_TEST_SHAS=&amp;JDK_BRANCH=jdk-17.0.6%2B8_adopt&amp;LABEL_ADDITION=&amp;ARTIFACTORY_REPO=&amp;ARTIFACTORY_ROOT_DIR=&amp;UPSTREAM_TEST_JOB_NUMBER=94&amp;DOCKERIMAGE_TAG=&amp;JDK_IMPL=hotspot&amp;TEST_TIME=&amp;SSH_AGENT_CREDENTIAL=&amp;AUTO_DETECT=true&amp;SLACK_CHANNEL=&amp;DYNAMIC_COMPILE=false&amp;ADOPTOPENJDK_SYSTEMTEST_OWNER_BRANCH=adoptium%3Amaster&amp;CUSTOMIZED_SDK_URL_CREDENTIAL_ID=eclipse_temurin_bot_email_and_token&amp;ARCHIVE_TEST_RESULTS=false&amp;NUM_MACHINES=&amp;OPENJDK_SHA=&amp;TRSS_URL=&amp;USE_TESTENV_PROPERTIES=true&amp;BUILD_LIST=openjdk&amp;UPSTREAM_JOB_NUMBER=6&amp;STF_OWNER_BRANCH=adoptium%3Amaster&amp;TIME_LIMIT=10&amp;JVM_OPTIONS=&amp;PARALLEL=None)

**Comment by Haroon-Khel** [comment]
Rerunning

build-alibaba-win2012r2-x64-2 https://ci.adoptopenjdk.net/job/Grinder/6358/console
test-ibmcloud-win2012r2-x64-1 https://ci.adoptopenjdk.net/job/Grinder/6359/console
test-azure-win2012r2-x64-3 https://ci.adoptopenjdk.net/job/Grinder/6360/console

**Comment by Haroon-Khel** [comment]
Passes on only. I'll take test-ibmcloud-win2012r2-x64-1 offline, but keep build-alibaba-win2012r2-x64-2 online as its a build machine with no test label

**Comment by sophia-guo** [comment]
a few other tests got the same issue only pass on test-azure-win2012r2-x64-3

[javax/net/ssl/SSLSocket/Tls13PacketSize.java.Tls13PacketSize](https://ci.adoptium.net/job/Test_openjdk21_hs_extended.openjdk_x86-64_windows_testList_1/19/testReport/junit/javax_net_ssl_SSLSocket_Tls13PacketSize/java/Tls13PacketSize/)
[security/infra/java/security/cert/CertPathValidator/certification/GoogleCA.java.GoogleCA](https://ci.adoptium.net/job/Test_openjdk21_hs_extended.openjdk_x86-64_windows_testList_1/19/testReport/junit/security_infra_java_security_cert_CertPathValidator_certification_GoogleCA/java/GoogleCA/)
[tools/jpackage/share/PerUserCfgTest.java.PerUserCfgTest](https://ci.adoptium.net/job/Test_openjdk21_hs_extended.openjdk_x86-64_windows_testList_1/19/testReport/junit/tools_jpackage_share_PerUserCfgTest/java/PerUserCfgTest/)
[tools/jpackage/share/ServiceTest.java.ServiceTest](https://ci.adoptium.net/job/Test_openjdk21_hs_extended.openjdk_x86-64_windows_testList_1/19/testReport/junit/tools_jpackage_share_ServiceTest/java/ServiceTest/)
[tools/jpackage/windows/Win8282351Test.java.Win8282351Test](https://ci.adoptium.net/job/Test_openjdk21_hs_extended.openjdk_x86-64_windows_testList_1/19/testReport/junit/tools_jpackage_windows_Win8282351Test/java/Win8282351Test/)
[tools/jpackage/windows/WinLongVersionTest.java#id1.WinLongVersionTest_id1](https://ci.adoptium.net/job/Test_openjdk21_hs_extended.openjdk_x86-64_windows_testList_1/19/testReport/junit/tools_jpackage_windows_WinLongVersionTest/java_id1/WinLongVersionTest_id1/)

**Comment by smlambert** [comment]
Since test-azure-win2012r2-x64-* machines have been replaced with test-azure-win2022-x64-* machines, we do not have any machines where this jdk_tools testcase can pass.

**Comment by smlambert** [comment]
@steelhead31 - could this be a similar issue to what you have seen where 'stuff' as a service on Windows machines has weird characters in the path?  I am noting the error msg (see https://github.com/adoptium/infrastructure/issues/2885#issuecomment-1890709946) that reads: `The following string format was not expected by the external UI message logger: "The Windows Installer Service could not be accessed. This can occur if the Windows Installer is not correctly installed. Contact your support personnel for assistance.".`

**Comment by steelhead31** [comment]
> @steelhead31 - could this be a similar issue to what you have seen where 'stuff' as a service on Windows machines has weird characters in the path? I am noting the error msg (see [#2885 (comment)](https://github.com/adoptium/infrastructure/issues/2885#issuecomment-1890709946)) that reads: `The following string format was not expected by the external UI message logger: "The Windows Installer Service could not be accessed. This can occur if the Windows Installer is not correctly installed. Contact your support personnel for assistance.".`

@smlambert its possible, but I think there are a couple of more likely culprits related to windows configuration / group policy settings, the later versions of windows lock down the windows installer service quite a lot, and some of the azure windows images have quite locked down permissions, if you'd like I can break this out into a separate issue, and I can investigate further?... It'd be handy if somebody could give me some pointers on the aqa targets required for me to be able to test this on a command line though!

**Comment by smlambert** [comment]
re: https://github.com/adoptium/infrastructure/issues/2885#issuecomment-1891644837

in Jenkins, TARGET=jdk_tools


Steps to run on outside of Jenkins / on commandline (presumes you have a Temurin JDK downloaded onto the machine into /jdkDir replace with actual location of your JDK that you want to test):

```
git clone https://github.com/adoptium/aqa-tests.git
cd aqa-tests
export TEST_JDK_HOME=/jdkDir
export BUILD_LIST=openjdk
export CUSTOM_TARGET=tools/jpackage/windows/Win8282351Test.java 
./get.sh
cd TKG
make compile
make _jdk_custom
```

For a more complete run of all testcases in the jdk_tools target, not just Win8282351Test, you could change the last line to:
```
make _jdk_tools
```

**Comment by steelhead31** [comment]
Test passed on win2022 machine 3 .

```
# AQACert.log content: 
# 
# Hostname: test-azure-win-2
#SHA.txt content: 
# 
# Timestamp: Tue Feb 13 11:42:48 2024 UTC 
# CUSTOM_TARGET: tools/jpackage/windows/Win8282351Test.java
# RESULTS_SUMMARY: TOTAL: 1 EXECUTED: 1 PASSED: 1 FAILED: 0 DISABLED: 0 SKIPPED: 0
1..1
ok 1 - jdk_custom_0
 ---
   duration_ms: 171050
 ...
```

**Comment by steelhead31** [comment]
Specific test, does pass as both adoptopenjdk and jenkins user on all 3 test-azure-win2022-x64-x machines, testing all tools tests... testing with jdk17.0.9+9.1

**Comment by steelhead31** [comment]
I've managed to replicate the error via grinder on one machine [https://ci.adoptium.net/job/Grinder/8812/console](https://ci.adoptium.net/job/Grinder/8812/console) , despite it passing locally.. so it definitely seems machine specific

**Comment by steelhead31** [comment]
Works ok via cli & jenkins on test-azure-win2022-x64-3 , works via cli on test-azure-win2022-x64-1 & test-azure-win2022-x64-2, however fails via grinder on both test-azure-win2022-x64-1 & test-azure-win2022-x64-2.. investigating further, suspect its a windows oddity.... as all 3 machines were built from the same playbooks.

**Comment by steelhead31** [comment]
This is now working on all 3 machines..  ( machine 1: [https://ci.adoptium.net/job/Grinder/8826/console](https://ci.adoptium.net/job/Grinder/8826/console) , machine 2 : [https://ci.adoptium.net/job/Grinder/8827/console](https://ci.adoptium.net/job/Grinder/8827/console) & machine 3 : [https://ci.adoptium.net/job/Grinder/8825/console](https://ci.adoptium.net/job/Grinder/8825/console) ).. .Thanks muchly!

**Comment by steelhead31** [comment]
I believe this to be the same root cause as the WSL detection issue, in that the PATH environment variable was being corrupted by the windows service wrapper used to run jenkins as a service, these machines were switched to using a CLI startup, and I believe this is the reason the tests now pass.

**Comment by karianna** [comment]
@steelhead31 So we can close this?

**Comment by steelhead31** [comment]
@karianna yes, I was just waiting on a final confirmation, but it all looks good now.
